adapters/pubmatic/pubmatic.go

```diff
@@ -424,6 +425,27 @@ func (a *PubmaticAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externa
 	return bidResponse, errs
 }
 
+func getNativeAdm(adm string) (string, error) {
+	var err error
+	nativeAdm := make(map[string]interface{})
+	err = json.Unmarshal([]byte(adm), &nativeAdm)
+	if err != nil {
+		return adm, err
+	}
+
+	// move bid.adm.native to bid.adm
+	if _, ok := nativeAdm["native"]; ok {
+		//using jsonparser to avoid marshaling, encode escape, etc.
+		value, _, _, err := jsonparser.Get([]byte(adm), string(openrtb_ext.BidTypeNative))
+		if err != nil {
+			return adm, err
+		}
+		adm = string(value)
+	}
```

Comment:
Updated the error messages.

Will add test case for line 441 later as I'm unable to mock `jsonparser.Get` for error scenarios since it would always get a valid json.